WordPress/Classes/Extensions/NSURL+Exporters.swift

```diff
@@ -0,0 +1,325 @@
+                let imgGenerator = AVAssetImageGenerator(asset: asset)
+                imgGenerator.maximumSize = targetSize
+                imgGenerator.appliesPreferredTrackTransform = true
+                let cgImage = try imgGenerator.copyCGImageAtTime(CMTimeMake(0, 1), actualTime: nil)
+                let uiImage = UIImage(CGImage: cgImage)
+                try uiImage.writeJPEGToURL(url)
+                successHandler(resultingSize: uiImage.size)
+            } catch let error as NSError {
+                errorHandler(error: error)
+            } catch {
+                errorHandler(error: errorForCode(.FailedToExport,
+                    failureReason: NSLocalizedString("Unknown asset export error", comment: "Error reason to display when the export of a image from device library fails")
+                    ))
+            }
+        } else {
+            errorHandler(error: errorForCode(.FailedToExport,
+                failureReason: NSLocalizedString("Unknown asset export error", comment: "Error reason to display when the export of a image from device library fails")
+                ))
+        }
+    }
+
+    func originalUTI() -> String? {
+        return typeIdentifier
+    }
+
+    var defaultThumbnailUTI: String {
+        get {
+            return kUTTypeJPEG as String
+        }
+    }
+
+    var assetMediaType: MediaType {
+        get {
+            if isImage {
+                return .Image
+            } else if (isVideo) {
+                return .Video
+            }
+            return .Document
+        }
+    }
+
+    //MARK: - Helper methods
+
+    var pixelSize: CGSize {
+        get {
+            if isVideo {
+                let asset = AVAsset(URL: self)
+                guard let track = asset.tracksWithMediaType(AVMediaTypeVideo).first else {
+                    return CGSizeZero
+                }
+                let size = CGSizeApplyAffineTransform(track.naturalSize, track.preferredTransform)
+                return size
+            } else if isImage {
+                let options: [NSString:NSObject] = [kCGImageSourceShouldCache: false]
+                guard
+                    let imageSource = CGImageSourceCreateWithURL(self, nil),
+                    let imageProperties = CGImageSourceCopyPropertiesAtIndex(imageSource, 0, options)
+                    else {
+                        return CGSizeZero
+                }
+
+                let imageDictionary = imageProperties as NSDictionary
+                guard
+                    let pixelWidth = imageDictionary[kCGImagePropertyPixelWidth as NSString] as? Int,
+                    let pixelHeight = imageDictionary[kCGImagePropertyPixelHeight as NSString] as? Int
+                    else {
+                        return CGSizeZero
+                }
+
+                return CGSize(width: pixelWidth, height: pixelHeight)
+            } else {
+                return CGSizeZero
+            }
+        }
+    }
```

Comment:
This getter (for `pixelSize` is a little hard to follow the flow because of all of the returns – you could perhaps restructure it a little so that there are only three:

```
var pixelSize: CGSize {
    get {
        if isVideo {
            let asset = AVAsset(URL: self)
            if let track = asset.tracksWithMediaType(AVMediaTypeVideo).first {
                return CGSizeApplyAffineTransform(track.naturalSize, track.preferredTransform)
            }
        } else if isImage {
            let options: [NSString:NSObject] = [kCGImageSourceShouldCache: false]
            if let imageSource = CGImageSourceCreateWithURL(self, nil),
                let imageDictionary = CGImageSourceCopyPropertiesAtIndex(imageSource, 0, options) as? NSDictionary,
                let pixelWidth = imageDictionary[kCGImagePropertyPixelWidth as NSString] as? Int,
                let pixelHeight = imageDictionary[kCGImagePropertyPixelHeight as NSString] as? Int {
                return CGSize(width: pixelWidth, height: pixelHeight)
            }
        }

        return CGSizeZero
    }
}
```